test/functional/wallet_importdescriptors.py

```diff
@@ -480,7 +480,7 @@ def run_test(self):
         addr = wmulti_pub.getnewaddress('', 'bech32')
         assert_equal(addr, 'bcrt1qp8s25ckjl7gr6x2q3dx3tn2pytwp05upkjztk6ey857tt50r5aeqn6mvr9') # Derived at m/84'/0'/0'/1
         change_addr = wmulti_pub.getrawchangeaddress('bech32')
-        assert_equal(change_addr, 'bcrt1qt9uhe3a9hnq7vajl7a094z4s3crm9ttf8zw3f5v9gr2nyd7e3lnsy44n8e')
+        assert_equal(change_addr, 'bcrt1qzxl0qz2t88kljdnkzg4n4gapr6kte26390gttrg79x66nt4p04fssj53nl')
```

Comment:
Perhaps the assertions below can be added.
They test that `importdescriptors` retrieved the mempool transaction.

Not related to this PR, but transaction `send_txid` is confirmed in line 457, isn't it ? 
Why is it still in the mempool in line 463 and 470?

```suggestion
        assert_equal(change_addr, 'bcrt1qzxl0qz2t88kljdnkzg4n4gapr6kte26390gttrg79x66nt4p04fssj53nl')
        assert(any([x == send_txid for x in self.nodes[0].getrawmempool(True)]))
        assert(any([x['txid'] == send_txid for x in wmulti_pub.listunspent(0)]))
```

Reply:
Added

> Not related to this PR, but transaction `send_txid` is confirmed in line 457, isn't it ? Why is it still in the mempool in line 463 and 470?

Hm, you are right that this shouldn't really work but it still does due to a race between the two nodes involved. I am kind of surprised that this doesn't lead flakiness in the CI but I couldn't find any related issue. I addressed this in #25476.